File: Software Design and Engineering/IT145_RescueAnimal_Enhancement/Monkey.py
from RescueAnimal import RescueAnimal

class Monkey(RescueAnimal):
    # Constructor method to initialize Monkey-specific attributes as well as common RescueAnimal attributes
    def __init__(self, name, species, gender, age, weight, acquisition_date, acquisition_country,
                 training_status, reserved, in_service_country, tail_length, height, body_length):
        # Call the RescueAnimal constructor to initialize common attributes
        super().__init__(name, "Monkey", gender, age, weight, acquisition_date, acquisition_country,
                         training_status, reserved, in_service_country)
        # Monkey-specific attributes
        self.species = species
        self.tail_length = tail_length
        self.height = height
        self.body_length = body_length

    # Getter and setter methods for Monkey-specific attributes
    def get_species(self):
        return self.species

    def set_species(self, species):
        self.species = species

    def get_tail_length(self):
        return self.tail_length

    def set_tail_length(self, tail_length):
        self.tail_length = tail_length

    def get_height(self):
        return self.height

    def set_height(self, height):
        self.height = height

    def get_body_length(self):
        return self.body_length

    def set_body_length(self, body_length):
        self.body_length = body_length

File: Software Design and Engineering/IT145_RescueAnimal_Original/Monkey.java

public class Monkey extends RescueAnimal {
	//Instance Fields
	private String tailLength;
	private String height;
	private String bodyLength;
	private String species;
	
	//Constructor
	public Monkey(String name, String species, String gender, String age,String weight, String acquisitionDate, String acquisitionCountry, 
	String trainingStatus, boolean reserved, String inServiceCountry, String tailLength, String height, String bodyLength) { 
        setSpecies(species);
        setTailLength(tailLength);
        setHeight(height);
        setBodyLength(bodyLength);
}
	
	//Accessors and Mutators
	public void setTailLength(String tailLength) {
		this.tailLength = tailLength;
	}
	public String getTailLength() {
		return tailLength;
	}
	public void setHeight(String height) {
		this.height = height;
	}
	public String getHeight() {
		return height;
	}
	public void setBodyLength(String bodyLength) {
		this.bodyLength = bodyLength;
	}
	public String getBodyLength() {
		return bodyLength;
	}
	public void setSpecies(String species) {
		this.species = species;
	}
	public String getSpecies() {
		return species;
	}
}
